server/src/main/java/uk/ac/cam/cl/echo/extrusionfinder/server/preprocessor/ColorSpaceConversion.java

```diff
@@ -0,0 +1,180 @@
+package uk.ac.cam.echo.extrusionfinder.preprocessor;
+
+import java.lang.Math;
+
+import uk.ac.cam.cl.echo.extrusionfinder.server.imagedata.LabImageData;
+import uk.ac.cam.cl.echo.extrusionfinder.server.imagedata.RGBImageData;
+import uk.ac.cam.cl.echo.extrusionfinder.server.imagedata.XYZImageData;
+
+/**
+ * Collection of static colorspace conversion methods.
+ * Includes pairwise conversions between RGB, XYZ and L*a*b*.
+ */
+public class ColorSpaceConversion {
+
+    // The CIE XYZ tristimulus values of the reference white point for D6500
+    // wikipedia.org/wiki/Illuminant_D65#Definition
+    private static final double X_N = 0.95047;
+    private static final double Y_N = 1.00000;
+    private static final double Z_N = 1.08833;
+
+
+    // This composition is reccomended
+    /**
+     * Convert RGBImageData to LabImageData, preserving color
+     * <p>
+     * @param name Comment
+     * @throws Type Comment
+     * @return Comment
+     */
+    public static LabImageData toLab(RGBImageData rgb) {
+        return toLab(toXYZ(rgb));
+    }
+
+    public static RGBImageData toRGB(LabImageData lab) {
+        return toRGB(toXYZ(lab));
+    }
+
+
+    // The f(t) function from wikipedia.org/wiki/Lab_color_space#Forward_transformation
+    private static double f(double t) {
+        if (t > 6*6*6 / (29*29*29D)) {
```

Comment:
You use a lot of unnamed constants. I don't know what the significance of the numbers are throughout the code.

Either make them static final fields, or comment very clearly as to what they mean.